### PR TITLE
feat(MD-3347): add drage select for ios

### PR DESCRIPTION
**Pull request by vcellu**
Somethings of note.
No autoscroll at end of collection view

### CODE REVIEW

File: ios/PaddedLabel.swift [hunk 1]
```diff
@@ -63,21 +58,78 @@ class PaddedLabel: UILabel, SelectionsListener {
...
+        let convertedFrame = convertLocalFrameToSelectionBandFrame(selectionBand)
+        if envelope.frame.intersects(convertedFrame) {
+          addToSelections()
         }
       }
     }
   }
 
+  fileprivate func convertLocalFrameToSelectionBandFrame(_ selectionBand: UIView) -> CGRect {
+    let convertedFrame = convert(self.frame, from: self.superview)// convert(self.frame, to: dataCollectionView) //dataCollectionView.convert(self.frame, to: selectionBand)
```
Comment (gruskal) [inline ios/PaddedLabel.swift:112]: Comments needed?

**Review by gruskal — COMMENTED** [review]
I like the drag box, it would be nice to have a way to drag-deselect - but I don't think we support this on QCS, so it's a none issue.
The lasso is also still an option in the popover, we're removing that now right?

**Review by gruskal — COMMENTED** [review]
Looks good, some minor code comments.
Also, 
1) I need to tap twice sometimes to get it to select and the dragbox to appear. 
2) I cannot select cells in the last column
3) Sometimes selecting using the dragbox in one column, will select cells in another column

**Review by gruskal — APPROVED** [review]

File: ios/SelectionsEngine.swift [hunk 2]
```diff
@@ -9,13 +9,68 @@ import Foundation
 
 class SelectionsEngine: NSObject {
   var selections = [String: String]()
+  var pendingSelections = Set<String>()
   var onSelectionsChanged: RCTDirectEventBlock?
+  var onConfirmSelections: RCTDirectEventBlock?
   var selectionListners = [SelectionsListener]()
+  var activeColumn = -1.0
```
[the remark this answers is not in the document]
Comment (vcellu) [reply ios/SelectionsEngine.swift:16]: I set it to floating point, so it doesn't have to be casted all the time.

**Pull request merged**